Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketConnection.cs

```diff
@@ -224,6 +232,30 @@ public async Task SendAsync()
         }
     }
 
+    private void CollectTelemetry()
+    {
+        if (Version < BoltProtocolVersion.V5_4)
+        {
+            // telemetry not supported before 5.4
+            return;
+        }
+
+        lock (_telemetryCollector)
+        {
+            // add a telemetry message if we have enough now
+            if (_messages.Any(m => m is RunWithMetadataMessage) && _telemetryCollector.BatchSizeReached)

```

Comment:
Should probably be TransactionStartingMessage that covers all Tx APIs and session.run

Reply:
I think talk about this when the spike becomes reality (I agree though).